limitador-server/src/main.rs

```diff
@@ -26,35 +30,48 @@ pub enum Limiter {
     Async(AsyncRateLimiter),
 }
 
+enum StorageType {
+    Redis { url: String },
+    Infinispan { url: String },
+    InMemory,
+}
+
 impl Limiter {
     pub async fn new() -> Limiter {
-        let rate_limiter = match env::var("REDIS_URL") {
-            Ok(redis_url) => {
-                let storage = Self::storage_using_redis(&redis_url).await;
-                let mut rate_limiter_builder = AsyncRateLimiterBuilder::new(storage);
-
-                if Self::env_option_is_enabled(LIMIT_NAME_IN_PROMETHEUS_LABELS_ENV) {
-                    rate_limiter_builder = rate_limiter_builder.with_prometheus_limit_name_labels()
-                }
+        let rate_limiter = match Self::storage_type() {
+            StorageType::Redis { url } => Self::redis_limiter(&url).await,
+            StorageType::Infinispan { url } => Self::infinispan_limiter(&url).await,
+            StorageType::InMemory => Self::in_memory_limiter(),
+        };
 
-                Limiter::Async(rate_limiter_builder.build())
-            }
-            Err(_) => {
-                // Default to in-memory storage implementation
+        rate_limiter.load_limits_from_file().await;
 
-                let mut rate_limiter_builder = RateLimiterBuilder::new();
+        rate_limiter
+    }
 
-                if Self::env_option_is_enabled(LIMIT_NAME_IN_PROMETHEUS_LABELS_ENV) {
-                    rate_limiter_builder = rate_limiter_builder.with_prometheus_limit_name_labels()
-                }
+    fn storage_type() -> StorageType {
+        let redis_url = env::var(REDIS_URL_ENV);
+        let infinispan_url = env::var(INFINISPAN_URL_ENV);
 
-                Limiter::Blocking(rate_limiter_builder.build())
+        match (redis_url, infinispan_url) {
+            (Ok(_), Ok(_)) => {
+                panic!("Please set the Redis or the Infinispan URL, but not both");
```

Comment:
Panicking is not (user) error management - you should exit the process nicely, not with a screenful of internal info and a backtrace.

Reply:
:+1:
It's using `thiserror` now.